pkg/controller/aws/network/vpc/controller.go

```diff
@@ -150,17 +151,41 @@ func (e *external) Create(ctx context.Context, mgd resource.Managed) (resource.E
 
 	cr.Status.SetConditions(runtimev1alpha1.Creating())
 
-	req := e.client.CreateVpcRequest(&awsec2.CreateVpcInput{
-		CidrBlock: aws.String(cr.Spec.CIDRBlock),
-	})
-	req.SetContext(ctx)
+	// if VPCID already exists, skip creating the vpc
+	// this happens when an error has occurred when modifying vpc attributes
+	if cr.Status.VPCID == "" {
```

Comment:
does this VPCID value get stored in the managed resource's status every time we `Observe`? Just wanting to make sure that since we're using this value for control flow now that it will get set again if the status is for some reason wiped out.  For example from [Kubernetes API conventions doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status):

>  the status could be reconstructed by observation if it were lost.

Reply:
VPCID is stored in the managed resource's status. However since `VPCID` gets generated only after creating a vpc, it is stored after creation. It then is used to checked whether the resource exists during `observer`. If the status is wiped out, the vpc gets recreated.

Reply:
@jbw976 A general theme with the AWS connectivity primitives is that they have non-deterministic identifiers that are generated by AWS at creation time. This is tricky because it means the resource don't comply with Kubernetes API conventions, but I can't see any obvious way around this.